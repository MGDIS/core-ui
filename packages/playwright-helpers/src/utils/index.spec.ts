import { describe, expect, test } from 'vitest';
import { renderAttributes, renderProperties } from '.';

describe('utils', () => {
  describe('renderAttributes', () => {
    test.each([
      { args: undefined, expected: '' },
      { args: null, expected: '' },
      { args: {}, expected: '' },
      { args: { name: 'batman' }, expected: 'name="batman"' },
      { args: { name: 'batman', user: 'bruce', active: true, id: 1, object: {}, array: [], function: () => {} }, expected: 'name="batman" user="bruce" active="true" id="1"' },
    ])('Should render attributes', ({ args, expected }) => {
      const res = renderAttributes(args);
      expect(res).toEqual(expected);
    });
  });

  describe('renderProperties', () => {
    test('Should render properties, case id selector', () => {
      const res = renderProperties({ object: {} }, '#id');
      expect(res).toEqual("\n  document.getElementById('id').object={}");
    });
    test.each([
      { args: undefined, expected: '' },
      { args: null, expected: '' },
      { args: {}, expected: '' },
      { args: { name: 'batman' }, expected: '\n  ' },
    ])('Should not render attributes as properties', ({ args, expected }) => {
      const res = renderProperties(args, 'div');
      expect(res).toEqual(expected);
    });
    test.each([undefined, () => {}].map(fn => ({ name: 'batman', user: 'bruce', active: true, id: 1, object: {}, array: [], fn })))(
      'Should render properties, case id selector',
      args => {
        const res = renderProperties(args, 'div');
        expect(res).toMatchSnapshot();
      },
    );
  });
});
