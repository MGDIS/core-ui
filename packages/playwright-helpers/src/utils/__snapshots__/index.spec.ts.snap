// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renderProperties > Should render properties, case id selector 1`] = `
"
  document.querySelector('div').object={};
document.querySelector('div').array=[]"
`;

exports[`renderProperties > Should render properties, case id selector 2`] = `
"
  document.querySelector('div').object={};
document.querySelector('div').array=[];
document.querySelector('div').fn=() => {\\n  }"
`;

exports[`utils > renderProperties > Should render properties, case id selector 1`] = `
"
  document.querySelector('div').object={};
document.querySelector('div').array=[]"
`;

exports[`utils > renderProperties > Should render properties, case id selector 2`] = `
"
  document.querySelector('div').object={};
document.querySelector('div').array=[];
document.querySelector('div').fn=() => {\\n    }"
`;
